cli/src/main/kotlin/ls/kafka/backup/s3/BackupBucket.kt

```diff
@@ -69,15 +73,20 @@ class BackupBucket(private val bucket: String, s3Config: S3Config, private val k
             ByteArraySerializer(),
             ByteArraySerializer()
         ).use { producer ->
-            backupFiles.collect {
-                it.records.forEach { record ->
+            var count = 0
+            backupFiles.withIndex().collect { (index, file) ->
+                count = index
+                if (index > 0 && index % RECORDS_PER_MESSAGE == 0) {
```

Comment:
I think we need to call `producer.flush()` like every batch in order to ensure that the producer queue is flushed. AFAIK `.send()` is asynchronous.
I think it's fine to to the logging on every flush then.